test/e2e-techpreview/onclusterlayering_test.go

```diff
@@ -1053,3 +1053,76 @@ func TestSSHKeyAndPasswordForOSBuilder(t *testing.T) {
 		t.Logf("Deleted MachineConfig %s", testConfig.Name)
 	})
 }
+
+func TestExtensionsForOSBuilder(t *testing.T) {
+	t.Helper()
+
+	cs := framework.NewClientSet("")
+
+	// label random node from pool, get the node
+	unlabelFunc := helpers.LabelRandomNodeFromPool(t, cs, "worker", "node-role.kubernetes.io/layered")
+
+	// prepare for on cluster build test
+	prepareForOnClusterLayeringTest(t, cs, onClusterLayeringTestOpts{
+		poolName:             layeredMCPName,
+		customDockerfiles:    map[string]string{},
+		useEtcPkiEntitlement: true,
+	})
+
+	testConfig := &mcfgv1.MachineConfig{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   "99-extensions",
+			Labels: helpers.MCLabelForRole(layeredMCPName),
+		},
+		Spec: mcfgv1.MachineConfigSpec{
+			Config: runtime.RawExtension{
+				Raw: helpers.MarshalOrDie(ctrlcommon.NewIgnConfig()),
+			},
+			Extensions: []string{"usbguard", "kerberos"},
+		},
+	}
+
+	helpers.SetMetadataOnObject(t, testConfig)
+
+	// Create the MachineConfig and wait for the configuration to be applied
+	_, err := cs.MachineConfigs().Create(context.TODO(), testConfig, metav1.CreateOptions{})
+	require.Nil(t, err, "failed to create MC")
+	t.Logf("Created MC %s", testConfig.Name)
+
+	// wait for rendered config to finish creating
+	renderedConfig, err := helpers.WaitForRenderedConfig(t, cs, layeredMCPName, testConfig.Name)
+	require.Nil(t, err)
+	t.Logf("Finished rendering config %s", renderedConfig)
+
+	// wait for mcp to complete updating
+	err = helpers.WaitForPoolComplete(t, cs, layeredMCPName, renderedConfig)
+	require.Nil(t, err)
+	t.Logf("Pool %s completed updating", layeredMCPName)
+
+	// Validate the extensions are installed
+	osNode := helpers.GetSingleNodeByRole(t, cs, layeredMCPName) // Re-fetch node with updated configurations
+
+	foundPkg := helpers.ExecCmdOnNode(t, cs, osNode, "rpm", "-q", "usbguard")
+	if strings.Contains(foundPkg, "package usbguard is not installed") {
+		t.Fatalf("usbguard extensions not found, got %s", foundPkg)
+	}
+	t.Logf("usbguard extension installed, got %s", foundPkg)
+
+	foundPkg = helpers.ExecCmdOnNode(t, cs, osNode, "rpm", "-q", "kerberos")
```

Comment:
note: `rpm -q` returns a non-zero exit code when the package is not found. Given how `helpers.ExecCmdOnNode()` is written, I don't think it will return a value when the program running on the host returns a non-zero exit code. Instead, the test will fail at that point (which is fine in this case, in my opinion).

Reply:
Fixed